### PR TITLE
Add some tests for Download class

### PR DESCRIPTION
**Pull request by dennisschagt**
Definitely doesn't test all behavior of the Download class but I would like to get some feedback before writing too many tests.

### CODE REVIEW

File: mk/mk.deps [hunk 1]
```diff
@@ -663,6 +663,7 @@ test/opmlurlreader.o: test/opmlurlreader.cpp include/opmlurlreader.h \
  include/configactionhandler.h include/urlreader.h 3rd-party/catch.hpp \
  test/test-helpers.h include/utils.h include/logger.h config.h \
  include/strprintf.h
+test/download.o: test/download.cpp include/download.h
```
Comment (Minoru) [inline mk/mk.deps]: This is missing `test/test-helpers.h`. Note that this file is updated by `make deps`, so you can just run that and commit the result.

We always forget to re-run that command, so don't worry if there are some unrelated changes—just commit the whole file.

File: test/download.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+		std::function<void()> callback = [&got_called]() {
+			got_called = true;
+		};
+
+		Download d(callback);
+
+		REQUIRE(d.status() == DlStatus::QUEUED);
+
+		WHEN("its progress is updated (increased)") {
+			got_called = false;
```
Comment (Minoru) [inline test/download.cpp]: There is no need to reset the variable: Catch2 re-runs the whole TEST_CASE in such a way that each given-when-then path is run exactly once. Same goes for `SECTION` macros: TEST_CASE is re-run such that each "leaf" section is visited only once.

Ditto for other resets below.

File: test/download.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+			d.set_status(DlStatus::QUEUED);
+
+			THEN("the require-view-update callback does not get called") {
+				REQUIRE_FALSE(got_called);
+			}
+		}
+	}
+}
+
+TEST_CASE("Filename returns configured filename", "[Download]")
```
Comment (Minoru) [inline test/download.cpp]: I'd prefer this to say "filename() returns download's target filename", because the current description is quite tautological and doesn't tell much (to me, at least).

File: test/download.cpp [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+{
+	auto emptyCallback = []() {};
+
+	Download d(emptyCallback);
+
+	SECTION("filename returns empty string by default") {
+		REQUIRE(d.filename() == "");
```
Comment (Minoru) [inline test/download.cpp]: Please use `empty()` here. (Classic nitpick, I know :)

File: mk/mk.deps (hunk 1 above)
Comment (dennisschagt) [reply mk/mk.deps]: Apparently the make target is called slightly differently: `make depslist`

Comment (Minoru) [reply mk/mk.deps]: Ah, yes, sorry about that—wrote the command from memory without checking.

**Comment by Minoru** [comment]
The new commits look good to me. Do you want this merged already, or do you plan to push more tests here? I guess it makes more sense to do the latter, but I'm okay with either approach.

**Comment by dennisschagt** [comment]
Either is fine with me so indeed probably best if I add some more tests to this PR before merging.
Not sure if I will get to it today.

**Comment by Minoru** [comment]
There's no hurry, so take your time ;) Tell me when you're added everything you wanted to, and I'll merge then.

**Comment by dennisschagt** [comment]
@Minoru The macOS builds are failing due to a single test failure:
```
test/rsspp_rssparser.cpp:38
...............................................................................
test/rsspp_rssparser.cpp:41: FAILED:
  REQUIRE( rsspp::RssParser::__w3cdtf_to_rfc822("-3") == "" )
with expansion:
  "Wed, 31 Dec 1969 15:59:59 +0000" == ""
```

The [build page](https://cirrus-ci.com/task/5236688679075840) shows `Auto upgraded to catalina-base!`. I expect the failure has something to do with that.
Do you have an idea of how to fix that?

---

Also, I think the amount of Download tests is fine for now (I don't see the value in adding tests for pure getter/setter functions like `url()`/`set_url()` and `kbps()`/`set_kbps()`).

Could you review the added tests?

I rebased onto a more current version of master to be able to see results on Coveralls.
From previous conversation it looked like force-pushing was only problematic when doing it multiple times in a PR.
Apparently even a single rebase/force-push doesn't guarantee an easy review (clicking `force-pushed` also shows changes in the new commits on master).
In future I will try not force-pushing at all during a PR (unless really necessary).

**Comment by Minoru** [comment]
> Do you have an idea of how to fix that?

I think I'll just add an `if` that checks for negative arguments and returns an empty string. I'll handle that.

> I don't see the value in adding tests for pure getter/setter functions like url()/set_url() and kbps()/set_kbps()

Agreed.

I'll do the review after fixing the Catalina problem. Thanks for the work!

**Comment by Minoru** [comment]
Forgot to reply to that one:

> I rebased onto a more current version of master to be able to see results on Coveralls.

I haven't fixed the coverage yet, so don't expect any reports for now. =\

**Comment by Minoru** [comment]
The Catalina problem is fixed BTW, so can you please rebase this onto master to fix the build here?

**Comment by dennisschagt** [comment]
Thanks, done

> > I rebased onto a more current version of master to be able to see results on Coveralls.
> 
> I haven't fixed the coverage yet, so don't expect any reports for now. =\

It looks like the coverage is correctly submitted to Coveralls though.

In Travis `Test Coverage` log a url is logged:
`{"message":"Job #1673.20","url":"https://coveralls.io/jobs/58403212"}`
from which I can reach overviews for specific files on this branch/build, e.g.:
[src/download.cpp coverage](https://coveralls.io/jobs/58403212/source_files/1556079639).

**Comment by Minoru** [comment]
> It looks like the coverage is correctly submitted to Coveralls though.

Only from PRs made from forks, and Coveralls doesn't post a comment with the results. All of that stuff used to work with cpp-coveralls, so I still consider coverage to be broken.

Looks like this is done, so I'll quickly skim the latest changes and merge already.

**Comment by Minoru** [comment]
Yup, looks great. Thank you for the work, @dennisschagt! :+1:
